data-es6.js

```diff
@@ -2914,6 +2914,30 @@ exports.tests = [
   },
 },
 {
+  name: 'prototype of bound functions',
+  category: 'functions',
+  significance: 'small',
+  link: 'https://people.mozilla.org/~jorendorff/es6-draft.html#sec-boundfunctioncreate',
+  exec: function () {/*
+    function getProtoBound(proto) {
+      var f = Object.setPrototypeOf(function() { }, proto)
+      var boundF = Function.prototype.bind.call(f, null)
+      return Object.getPrototypeOf(boundF)
+    }
+
+    return [ Function.prototype, function(){}, [], null ].every(function(proto) {
```

Comment:
I also think breaking those into subtests would be an overkill. Seems unlikely that one would work and not the other.

Sent from my iPhone

> On 05 Jun 2015, at 13:10, Claude Pache <notifications@github.com> wrote:
> 
> In data-es6.js:
> 
> > @@ -2914,6 +2914,30 @@ exports.tests = [
> >    },
> >  },
> >  {
> > +  name: 'prototype of bound functions',
> > +  category: 'functions',
> > +  significance: 'small',
> > +  link: 'https://people.mozilla.org/~jorendorff/es6-draft.html#sec-boundfunctioncreate',
> > +  exec: function () {/*
> > +    function getProtoBound(proto) {
> > +      var f = Object.setPrototypeOf(function() { }, proto)
> > +      var boundF = Function.prototype.bind.call(f, null)
> > +      return Object.getPrototypeOf(boundF)
> > +    }
> > +
> > +    return [ Function.prototype, function(){}, [], null ].every(function(proto) {
> I fear that adding subtests for other types of functions (arrows, generators, subclasses, etc.) would add more distraction than useful information. Unless the tested implementation introduces a strange bug in its modified version of .bind(), the result of the subtest for .bind on generators would just be the intersection of the result of the base test for .bind() and the one for the existence of generators; and a negative result of that subtest could only be interpreted correctly when compared with the existence of generators.
> 
> —
> Reply to this email directly or view it on GitHub.
>